lib/vtls/bearssl.c

```diff
@@ -792,19 +792,11 @@ static void *Curl_bearssl_get_internals(struct ssl_connect_data *connssl,
 static void Curl_bearssl_close(struct connectdata *conn, int sockindex)
 {
   struct ssl_connect_data *connssl = &conn->ssl[sockindex];
-  unsigned char *buf;
-  size_t len, i;
-  ssize_t ret;
+  size_t i;
 
   if(BACKEND->active) {
     br_ssl_engine_close(&BACKEND->ctx.eng);
-    while(br_ssl_engine_current_state(&BACKEND->ctx.eng) & BR_SSL_SENDREC) {
-      buf = br_ssl_engine_sendrec_buf(&BACKEND->ctx.eng, &len);
-      ret = swrite(conn->sock[sockindex], buf, len);
-      if(ret < 0)
-        break;
-      br_ssl_engine_sendrec_ack(&BACKEND->ctx.eng, ret);
-    }
+    (void)bearssl_run_until(conn, sockindex, BR_SSL_CLOSED);
```

Comment:
Is there a risk that this could hang? I want to avoid something like https://github.com/curl/curl/issues/4487#issuecomment-547293283

Reply:
If the socket is non-blocking, it will run until it receives a close-notify or error. Otherwise, it will run until close-notify, error, or EAGAIN on sread/swrite.

What are the exact requirements for the vtls close_one function?

Reply:
The sockets are non-blocking (I'm pretty sure this is all or most of the time) so curl can work on multiple transfers simultaneously. Functions should not block. As in the issue I referenced if it blocks waiting for a close_notify from the server then that is not acceptable.

Reply:
If the sockets are non-blocking, then `bearssl_run_until` will not block. (sorry, I originally said the reverse of what is true above)